module/os/freebsd/zfs/zfs_znode.c

```diff
@@ -464,8 +471,6 @@ zfs_znode_alloc(zfsvfs_t *zfsvfs, dmu_buf_t *db, int blksz,
 	atomic_store_ptr(&zp->z_cached_symlink, NULL);
 #endif
 
-	vp = ZTOV(zp);
```

Comment:
If that's not obvious, this is a reverse of the assignment on line 448 as `ZTOV` expands to `zp->z_vnode`.

Reply:
That would be this static analyzer warning:

https://pvs-studio.com/en/docs/warnings/v587/

If run another scan of the FreeBSD kernel while this is in -CURRENT, they probably would include this in a blog post.

None of LLVM/Clang, the Clang Static Analyzer or Clang-Tidy issue a warning about this code, so I filed an issue with LLVM:

https://github.com/llvm/llvm-project/issues/61229

Reply:
This was also filed with CodeQL:

https://github.com/github/codeql/issues/12417